### DISCUSSION TITLE
Missing request extension.

### DISCUSSION THREAD

**Discussion by abhaythakur71181** [tokio-rs/axum#2922] [Q&A] [answered]
### Summary

I am getting this error - 
```Missing request extension: Extension of type `alloc::sync::Arc<aniscraper::hianime::HiAnimeRust>` was not found. Perhaps you forgot to add it? See `axum::Extension`.```

but i have added extensions

```rust

    let hianime_cache = Arc::new(Mutex::new(HiAnime::new()));
    let hianime = Arc::new(HiAnimeRust::new(Some(secret_config)));
    let router = Router::new()
        .layer(Extension(hianime))
        .layer(Extension(hianime_cache));
```

### axum version

0.7.5

**Comment by abhaythakur71181** [comment]
impl HiAnimeRust {
    pub async fn new(secret: Option<SecretConfig>) -> Self {}

**Reply by jplatte** [reply, marked as answer]
That constructor is async, but you don't use `.await` wegen calling it, so you're adding a future to the request extensions rather than the object it produces.

**Reply by abhaythakur71181** [reply]
> That constructor is async, but you don't use `.await` wegen calling it, so you're adding a future to the request extensions rather than the object it produces.

sometimes my brain stops braining 😅
